org.jacoco.core/src/org/jacoco/core/analysis/Analyzer.java

```diff
@@ -108,6 +108,7 @@ public void visitEnd() {
 	public void analyzeClass(final ClassReader reader) {
 		final ClassVisitor visitor = createAnalyzingVisitor(
 				CRC64.checksum(reader.b), reader.getClassName());
+		// TODO maybe we can SKIP_FRAMES here?
```

Comment:
@marchof I believe that in this case `FrameNode` won't be appear in `methodNode.instructions` and since they anyway not used for analysis, we won't be wasting CPU cycles to skip them in filters - see method `SynchronizedFilter.Matcher#next()`. While this doesn't play any role for `SynchronizedFilter`, the same method is used in filter for try-with-resources (#500). And now that you reminded me about that - probably I can demonstrate observable impact on performance using its implementation prior to the last optimizations. I'm fine with postponing this, however this shouldn't be forgotten and in fact I almost forgot about this, but good that left this comment.